File: Anul 2/Semestrul 2/Metode de proiectare si programare/labs/Tema 1-proiect Java/proiect/src/main/java/com/example/proiect/repository/IRepository.java
package com.example.proiect.repository;

import com.example.proiect.exceptions.RepoException;

public interface IRepository<T, ID> {
    T add(T entity) throws RepoException;
    T remove(ID id) throws RepoException;
    T findElement(ID id) throws RepoException;
    Iterable<T> getAll();
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 2-proiect Csharp/proiect/proiect/repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using proiect.model;

namespace proiect.repository
{
    interface IRepository <ID, T> where T : Entity<ID>
    {
        T add(T entity);
        T remove (ID id);
        T findElement(ID id);
        IEnumerable<T> getAll();
    }
}
